panther_analysis_tool/schemas.py

```diff
@@ -57,68 +58,73 @@
     ignore_extra_keys=False,
 )
 
-POLICY_SCHEMA = Schema(
-    {
-        "AnalysisType": Or("policy"),
-        "Enabled": bool,
-        "Filename": str,
-        "PolicyID": And(str, NAME_ID_VALIDATION_REGEX),
-        "ResourceTypes": [str],
-        "Severity": Or("Info", "Low", "Medium", "High", "Critical"),
-        Optional("ActionDelaySeconds"): int,
-        Optional("AutoRemediationID"): str,
-        Optional("AutoRemediationParameters"): object,
-        Optional("Description"): str,
-        Optional("DisplayName"): And(str, NAME_ID_VALIDATION_REGEX),
-        Optional("OutputIds"): [str],
-        Optional("Reference"): str,
-        Optional("Runbook"): str,
-        Optional("Suppressions"): [str],
-        Optional("Tags"): [str],
-        Optional("Reports"): {str: list},
-        Optional("Tests"): [
-            {
-                "Name": str,
-                Optional(
-                    "ResourceType"
-                ): str,  # Not needed anymore, optional for backwards compatibility
-                "ExpectedResult": bool,
-                "Resource": object,
-            }
-        ],
-    },
-    ignore_extra_keys=False,
-)  # Prevent user typos on optional fields
 
-RULE_SCHEMA = Schema(
-    {
-        "AnalysisType": Or("rule"),
-        "Enabled": bool,
-        "Filename": str,
-        "RuleID": And(str, NAME_ID_VALIDATION_REGEX),
-        "LogTypes": [str],
-        "Severity": Or("Info", "Low", "Medium", "High", "Critical"),
-        Optional("Description"): str,
-        Optional("DedupPeriodMinutes"): int,
-        Optional("DisplayName"): And(str, NAME_ID_VALIDATION_REGEX),
-        Optional("OutputIds"): [str],
-        Optional("Reference"): str,
-        Optional("Runbook"): str,
-        Optional("SummaryAttributes"): [str],
-        Optional("Suppressions"): [str],
-        Optional("Threshold"): int,
-        Optional("Tags"): [str],
-        Optional("Reports"): {str: list},
-        Optional("Tests"): [
-            {
-                "Name": str,
-                Optional(
-                    "LogType"
-                ): str,  # Not needed anymore, optional for backwards compatibility
-                "ExpectedResult": bool,
-                "Log": object,
-            }
-        ],
-    },
-    ignore_extra_keys=False,
-)  # Prevent user typos on optional fields
+def get_rule_policy_schema(ignore_extra_keys: bool = False) -> Tuple[Schema, Schema]:
```

Comment:
I see, this is the workaround you were talking about? It is a bit heavy.

I know it's not python best practice, but it seems we can just refer to the internal `_ignore_extra_keys` field to set this value. Little example I whipped up in an interpreter:

```python
>>> from schema import Schema
>>> thing = Schema({"Type": str,}, ignore_extra_keys=True)
>>> thing.validate({'Type': 'hello', 'Other': 'Meme'})
{'Type': 'hello'}
>>> thing._ignore_extra_keys  = False
>>> thing.validate({'Type': 'hello', 'Other': 'Meme'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/schema.py", line 415, in validate
    raise SchemaWrongKeyError(message, e)
schema.SchemaWrongKeyError: Wrong key 'Other' in {'Type': 'hello', 'Other': 'Meme'}
```

So it seems we could leave this mostly the way it was before, and when the `--ignore-extra-fields` flag is set we can update the `POLICY_SCHEMA._ignore_extra_keys` variable. The `_` prefix means by convention we shouldn't interact with this directly. But for such a simple flag it seems almost like an oversight from the library to not make it modifiable, not that modifying it will have some strange behavior. Interested to hear what you think.

Reply:
That's doable, it'll throw a linting issue but we could ignore it.